### PR TITLE
Change ConvertToReflectometryQ to work with detectors at twoTheta

### PR DESCRIPTION
**Pull request by gemmaguest**
**Report to:** Max and Andrew at ISIS

Change `ConvertToReflectometryQ` to work with detectors positioned at twoTheta rather than theta.
 - The code change involves updating the function `setThetaFinal` to set thetaFinal to `twoTheta` minus the incident theta. Previously it set it to `twoTheta`.
 - To version the algorithm, a `version` member was required to be added to the underlying `ReflectometryTransform` class.
 - The only system test we have for this code is a POLREF test which is out of date. I have not been able to get scientists to provide a new test so I have updated the old test with the new results (modifying it slightly to give a slightly more sensible test case, although it still not great from a scientific perspective; however at least it will catch changes in the results). Scientists have agreed they would like this merged despite not having provided validation data. 

**To test:**

Run this script. The QxQz results should give a similar looking plot with v2 of the algorithm if you use the IDF from PR #30246 (which I've named INTER_Definition_2021 in the script below but you'll need to save this somewhere yourself to use it). Note that this is not a real world example so the workspaces are not exactly the same if you compare them, but the plots would be vastly different if you use the wrong IDF so this is a good indication it is working sensibly. Scientists are expecting a breaking change and will validate the new results properly during beta testing.
```
name='INTER58920'
ws1=Load(Filename=name)
LoadInstrument(Workspace=ws1, RewriteSpectraMap=False, Filename='INTER_Definition_2020.xml')
ConvertUnits(InputWorkspace=ws1, OutputWorkspace=name+'_lam1', Target='Wavelength')
ConvertSpectrumAxis(InputWorkspace=name+'_lam1', OutputWorkspace=name+'_det_thlam1', Target='SignedTheta')
ConvertToReflectometryQ(InputWorkspace=name+'_det_thlam1', Method='NormalisedPolygon', OverrideIncidentTheta=True, 
  IncidentTheta=0.5, OutputWorkspace=name+'_QxQz1', OutputVertexes=name+'_vertices',Extents='-0.0001, 0.0001, 0.01, 0.08',
  Version=1)

ws2=Load(Filename=name)
LoadInstrument(Workspace=ws2, RewriteSpectraMap=False, Filename='INTER_Definition_2021.xml')
ConvertUnits(InputWorkspace=ws2, OutputWorkspace=name+'_lam2', Target='Wavelength')
ConvertSpectrumAxis(InputWorkspace=name+'_lam2', OutputWorkspace=name+'_det_thlam2', Target='SignedTheta')
ConvertToReflectometryQ(InputWorkspace=name+'_det_thlam2', Method='NormalisedPolygon', OverrideIncidentTheta=True, 
  IncidentTheta=0.5, OutputWorkspace=name+'_QxQz2', OutputVertexes=name+'_vertices',Extents='-0.0001, 0.0001, 0.01, 0.08',
  Version=2)
```

Screenshot_2021-08-24_14-01-13

Fixes #20327

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by StephenSmith25** [comment]
Is it intended for this PR to make it into the 6.1 release?

**Comment by gemmaguest** [comment]
> 
> 
> Is it intended for this PR to make it into the 6.1 release?

No, sorry - I've changed the milestone

File: docs/source/algorithms/ConvertToReflectometryQ-v2.rst [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    MoveInstrumentComponent(Workspace = I, ComponentName = "lineardetector", X = 0, Y = 0, Z = zOffset)
+
+    # Should now have signed theta vs Lambda
+    ConvertSpectrumAxis(InputWorkspace=I,OutputWorkspace='SignedTheta_vs_Wavelength',Target='signed_theta')
+
+    qxqy, vertexes_qxqy = ConvertToReflectometryQ(InputWorkspace='SignedTheta_vs_Wavelength', OutputDimensions='Q (lab frame)', Extents='-0.0005,0.0005,0,0.12', OutputAsMDWorkspace=False,Method='NormalisedPolygon')
+
+    kikf, vertexes_kikf = ConvertToReflectometryQ(InputWorkspace='SignedTheta_vs_Wavelength', OutputDimensions='K (incident, final)', Extents='0,0.05,0,0.05', OutputAsMDWorkspace=False,Method='NormalisedPolygon')
+
+    pipf, vertexes_pipf = ConvertToReflectometryQ(InputWorkspace='SignedTheta_vs_Wavelength', OutputDimensions='P (lab frame)', Extents='0,0.1,-0.02,0.15', OutputAsMDWorkspace=False,Method='NormalisedPolygon')
```
Comment (Harrietbrown) [inline docs/source/algorithms/ConvertToReflectometryQ-v2.rst:131]: This example code is identical to the example code in the v1 doc. should they de different to show the difference between the versions?

**Review by Harrietbrown — APPROVED** [review]
The difference between workspaces with each instrument definition  is less than 0.1%. It is deaddy to go.

**Pull request merged**